src/libraries/Microsoft.PowerFx.Connectors/Shared/ServiceFunction.cs

```diff
@@ -425,26 +440,59 @@ public bool TryGetParamDefaultValue(string paramName, out Tuple<string, DType> d
 
         public async Task<FormulaValue> InvokeAsync(FormattingInfo context, FormulaValue[] args, CancellationToken cancellationToken)
         {
-            if (_invoker == null)
+            cancellationToken.ThrowIfCancellationRequested();
+
+            FormulaValue result = await (_invoker ?? throw new InvalidOperationException($"Function {Name} can't be invoked.")).InvokeAsync(context, args, cancellationToken).ConfigureAwait(false);
+            ExpressionError er = null;
+
+            if (result is ErrorValue ev && (er = ev.Errors.FirstOrDefault(e => e.Kind == ErrorKind.Network)) != null)
+            {
+                result = FormulaValue.NewError(new ExpressionError() { Kind = er.Kind, Severity = er.Severity, Message = $"{Namespace.ToDottedSyntax()}.{Name} failed: {er.Message}" }, ev.Type);
+            }
+
+            if (result is RecordValue rv && IsPageable && rv.TryGetField(FormulaType.String, _pageLink, out FormulaValue pageLink))
+            {
+                string nextLink = (pageLink as StringValue)?.Value;
+                result = new PagedRecordValue(rv, () => GetNextPage(nextLink, cancellationToken), _maxRows);
+            }
+
+            return result;
+        }
+
+        private FormulaValue GetNextPage(string nextLink, CancellationToken cancellationToken)
```

Comment:
>FormulaValue [](http://example.com/codeflow?start=16&length=12)

can this be more strongly typed? Record or Table?

Reply:
Could be a RecordValue

Reply:
After looking at the rest of the PR, 
could we do something even more strongly typed, like 
class Segment  { 
  TableValue value;
  string nextLink;
}

Essentially, consolidate all the "decoding" from RecordValue in one place. 
That may reduce some of the duplication and simplify the logic in Page*Value classes.

Reply:
I modifed & simplified the code to have Segment=PagedRecordValue